app/javascript/flavours/polyam/styles/components/notification.scss

```diff
@@ -486,12 +485,25 @@
           color: $primary-text-color;
         }
 
+        // Polyam: Upstream only sets color here. Other properties are to match behaviour in status.
         time {
           color: $dark-text-color;
+          text-align: end; // Polyam: Fixes wrapped timestamp leaving gap
+          white-space: nowrap; // Polyam: Prevent wrapping
+        }
+
+        // Polyam: Align timestamp to end
+        &-separator {
+          text-align: end;
+          flex-grow: 1;
         }
 
-        // Polyam: Removed time as it should be visible.
         @container (width < 350px) {
+          // Polyam: Prevent reflow on timestamp change
+          display: grid;
+          grid-template-columns: 1fr minmax(10%, max-content);
```

Comment:
This feels a bit awkward, but does indeed fix the reflow issue.
I'm not sure this works with every locale, but it did with a few different samples.